src/main/java/decodex/modules/binary/BinaryDecoder.java

```diff
@@ -0,0 +1,78 @@
+package decodex.modules.binary;
+
+import decodex.modules.Module;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import decodex.data.Data;
+import decodex.data.exception.ModuleException;
+
+/**
+ * The BinaryDecoder class handles the binary decoding operations.
+ */
+public class BinaryDecoder extends Module {
+
+    public static final String MODULE_NAME = "bindecode";
+    public static final String MODULE_DESCRIPTION = "Decodes the data from binary.";
+
+    private static final int BINARY_RADIX = 2;
+    private static final String VALID_BINARY_REGEX = "^([01]{8})*$";
+
+    public BinaryDecoder() {
+        super(MODULE_NAME, MODULE_DESCRIPTION);
+    }
+
+    /**
+     * Performs binary decoding on the provided Data object.
+     *
+     * @param data The Data to be decoded.
+     * @return A Data object containing the binary decoded data.
+     * @throws ModuleException The Data is not in binary format.
+     */

```

Comment:
Perhaps add the word `If` after the name of the exception? (See Tian Boon's convention https://github.com/AY2122S1-CS2113T-T10-3/tp/blob/master/src/main/java/decodex/data/DataManager.java#L30)